templates/cluster-template-custom-builds.yaml

```diff
@@ -0,0 +1,309 @@
+apiVersion: cluster.x-k8s.io/v1alpha4
```

Comment:
putting this template in templates/ means that this template will be discoverable as a flavor by `clusterctl --flavor` and included in auto-generated release artifacts. Given that it's intended for testing / dev purposes only, consider moving it to the test folder or some other specifc tilt dev folder. See my related comment [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1264#discussion_r602578491)

Reply:
So, like, add a new directory under `test/e2e/data` and then add a file that starts with `cluster_template_` that contains this entire flavor-generated yaml file?

Reply:
`templates/test` is what I was referring to

I suggest adding a new directory `templates/test/tilt`, moving the entire flavor there, and making sure that Tilt gets the templates in that directory as well here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/Tiltfile#L226

Reply:
if adding a new dir you also need to add it to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/hack/gen-flavors.sh#L25 so templates are generated by `make generate-flavors`

Reply:
@jackfrancis did you figure out how to do this? If not let's pair on it sometime this week so we can get this merged and unblock @chewong for #1264